### ISSUE TITLE
Ionic V4 Page Navigation Jankiness with 4.0.0-alpha-11

### ISSUE THREAD

**Issue by kinggolf** [ionic-team/ionic#14841]
The 1st page of my app is a list of events and the 2nd page shows details of the selected event when tapped by a user. I am use Cloud Firestore as the back end and accessing with AngularFIre2. When I upgraded from ionic/angular 4.0.0-alpha-7 to 4.0.0-alpha-11 this page navigation from events to details exhibited severe transition jankiness. The transition was is smooth with alpha-7. Also, tried alpha-9 and saw same jankiness.
I've tried to replicate this with a simplified project using mocked data, but the page transitions are smooth with all version of ionic/angular. I do not know if it this is due to the quantity of data or the inner workings of the Firestore query, or what.

I see this only on iOS. Android is fine even with 4.0.0-alpha-11.

Here is my Firestore query:
```
  getPlayersByGroup(eventIsHS, eventInfo) {
    const eventPlayersRef: AngularFirestoreDocument<SLPlayer> = this.af.doc(APPEVENTTYPE + 'Events/' + eventInfo.id);
    return eventPlayersRef.collection<SLPlayer>('players', p => p.orderBy('startHole').
    orderBy('group').orderBy('scOrder')).snapshotChanges().pipe(
      map(actions => actions.map(a => {
        const data = a.payload.doc.data() as SLPlayer;
        const id = a.payload.doc.id;
        if (!eventInfo.shotgunStart) {
          const teeTimeString = data.teeTime.toString(10);
          data.teeHours = teeTimeString.substr(0, teeTimeString.length - 2);
          data.teeMinutes = teeTimeString.substr(teeTimeString.length - 2);
        }
        return {id, ...data};
      }))
    );
  }
```
And here is the code in ngOnInit of the 2nd page component that calls the above method and then subscribes to it.
```
      this.playersByGroup$ = this.fsService.getPlayersByGroup(this.eventIsHS, this.eventInfo);
      this.playersByGroupSub = this.playersByGroup$.subscribe(players => {
        if (players) {
          this.store.dispatch({type: PLAYERS_BY_NAME, payload: players});
          this.showLoading = false;
          ...
```

If I remove the subscribe code, then there is no problem with page transition, but of course there is no data either!
This feels more like a support question, however this issue does not appear with 4.0.0-alpha-7, which is why I'm posting here.

**Comment by kinggolf** [comment]
Here's an update after more experimentation. 
First of all, I see this both on the iOS simulator as well as a real iPhone6s device.

I have an ion-spinner in the 2nd page template, as such:
```
<ion-content class="icview" style="margin-top: 12px;">
  <ion-spinner *ngIf="showLoading"></ion-spinner>
  <ion-list>
   ...
```
In the 2nd page component `this.showLoading = false;` removes the spinner in the view. However, If I put a 500ms time delay on this statement:
```
        setTimeout(() => {
          this.showLoading = false;
        }, 500);
```
then I see nice smooth transition with 4.0.0-alpha-11 in iOS. I can also simply remove the spinner in the template and achieve the same. I feel better about this whole thing now, but still do not understand why this helps in 4.0.0-alpha-11 & was unnecessary in 4.0.0-alpha-7.

Maybe somebody can make some sense of this - I hope anyway!
BTW - nothing else in package.json is being changed in this analysis, only ionic/angular. For thoroughness here are the dependencies.
```
"dependencies": {
    "@angular/common": "6.0.9",
    "@angular/core": "6.0.9",
    "@angular/forms": "6.0.9",
    "@angular/http": "6.0.9",
    "@angular/platform-browser": "6.0.9",
    "@angular/platform-browser-dynamic": "6.0.9",
    "@angular/router": "6.0.9",
    "@ionic-native/core": "5.0.0-beta.13",
    "@ionic-native/device": "5.0.0-beta.13",
    "@ionic-native/dialogs": "5.0.0-beta.13",
    "@ionic-native/file": "5.0.0-beta.13",
    "@ionic-native/ionic-webview": "5.0.0-beta.13",
    "@ionic-native/keyboard": "5.0.0-beta.13",
    "@ionic-native/splash-screen": "5.0.0-beta.13",
    "@ionic-native/status-bar": "5.0.0-beta.13",
    "@ionic/angular": "4.0.0-alpha.11",
    "@ionic/ng-toolkit": "1.0.0-rc.11",
    "@ionic/schematics-angular": "1.0.0-rc.11",
    "@ngrx/store": "6.0.1",
    "angularfire2": "5.0.0-rc.11",
    "cordova-android": "7.1.1",
    "cordova-ios": "4.5.4",
    "cordova-plugin-device": "2.0.2",
    "cordova-plugin-ionic-keyboard": "2.1.2",
    "cordova-plugin-ionic-webview": "2.0.0",
    "cordova-plugin-splashscreen": "5.0.2",
    "cordova-plugin-whitelist": "1.3.3",
    "core-js": "2.5.7",
    "firebase": "5.3.0",
    "promise-polyfill": "8.0.0",
    "rxjs": "6.2.2",
    "zone.js": "0.8.26"
  },
  "devDependencies": {
    "@angular/cli": "6.0.8",
    "@angular/compiler": "6.0.9",
    "@angular/compiler-cli": "6.0.9",
    "@angular/language-service": "6.0.9",
    "@angular-devkit/architect": "0.7.0-rc.1",
    "@angular-devkit/build-angular": "0.7.0-rc.1",
    "@angular-devkit/core": "0.7.0-rc.1",
    "@angular-devkit/schematics": "0.7.0-rc.1",
    "@types/jasmine": "~2.8.6",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~10.5.2",
    "codelyzer": "~4.4.2",
    "jasmine-core": "~3.1.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~2.0.2",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.1",
    "karma-jasmine": "~1.1.1",
    "karma-jasmine-html-reporter": "^1.2.0",
    "protractor": "~5.4.0",
    "ts-node": "~7.0.0",
    "tslint": "5.11.0",
    "typescript": "2.7.2"
  },
```

**Comment by kinggolf** [comment]
This still exists on V4.0.0-beta.7. I did some experimentation and found that by putting the ion-spinner inside ion-list tab the jankiness was cleaned up. Not sure why, but just going with it.
Not good:
```
<ion-content class="icview" style="margin-top: 12px;">
  <ion-spinner *ngIf="showLoading"></ion-spinner>
  <ion-list>
   ...
```

All good!
```
<ion-content class="icview" style="margin-top: 12px;">
  <ion-list>
     <ion-spinner *ngIf="showLoading"></ion-spinner>
   ...
```

**Comment by paulstelzer** [comment]
So I think we can close it, correct?

**Comment by kinggolf** [comment]
Yes thank you. I have two apps released and working fine: one on 4.0.0-beta.13 and the other on 4.0.0-beta.15. Both look good on 4.0.0-beta.16 as I continue making enhancements.
